chain/sub/incoming.go

```diff
@@ -350,22 +350,16 @@ func (mv *MessageValidator) Validate(ctx context.Context, pid peer.ID, msg *pubs
 		)
 		recordFailure(ctx, metrics.MessageValidationFailure, "add")
 		switch {
-		case xerrors.Is(err, messagepool.ErrSoftValidationFailure):
+		case xerrors.Is(err, messagepool.ErrMessageTooBig):
 			fallthrough
-		case xerrors.Is(err, messagepool.ErrRBFTooLowPremium):
+		case xerrors.Is(err, messagepool.ErrMessageValueTooHigh):
 			fallthrough
-		case xerrors.Is(err, messagepool.ErrTooManyPendingMessages):
+		case xerrors.Is(err, messagepool.ErrNotEnoughFunds):
 			fallthrough
-		case xerrors.Is(err, messagepool.ErrNonceGap):
-			fallthrough
-		case xerrors.Is(err, messagepool.ErrGasFeeCapTooLow):
-			fallthrough
-		case xerrors.Is(err, messagepool.ErrNonceTooLow):
-			fallthrough
-		case xerrors.Is(err, messagepool.ErrExistingNonce):
-			return pubsub.ValidationIgnore
-		default:
+		case xerrors.Is(err, messagepool.ErrInvalidToAddr):
```

Comment:
I'm not sure if any of these should be "reject" as they can happen during forks/syncing.

Reply:
To be clear, this is exactly the current behavior. Below are the errors we mark as reject:
ErrMessageTooBig
ErrMessageValueTooHigh
ErrNotEnoughFunds
ErrInvalidToAddr

Are you saying none of these should be marked as reject? @Stebalien

Reply:
Yeah, I know. I'm saying that they only "too big" should likely be an error. But we should check this with @arajasek first.

Reply:
I think they're all fine to treat as REJECT, except `ErrNotEnoughFunds`.

`ErrMessageValueTooHigh` is trying to send more than 2B FIL, while `ErrInvalidToAddr` is just checking for non-nil recipient, both of which seem fine to REJECT.

Agreed that `ErrNotEnoughFunds` should be an IGNORE, though.

Reply:
Ah, I see. Sg.

Reply:
Curious, why would ErrNotEnoughFunds be an IGNORE? Wouldn't you expect the sender to have done some basic validation before sending out the msg?

Reply:
Whether or not there are enough available funds depends on the tipset against which the message is checked, other messages in the message pool, etc.

Reply:
I see, so you could have funds available when the txn was created but subsequently created txns which also used up the funds could have been executed first leaving insufficient funds for the first one. Is that right?

Reply:
Pretty sure that we want ErrNotEnoughFunds to be reject, otherwise I can just spam the network with RBF messages from an address with barely any fil

Reply:
@magik6k But none of those messages would actually get added to anyone's mempools (assuming you don't have the FIL for it). You'll just spam and get ignored?

Sounds like you're maybe saying `ErrRBFTooLowPremium` should be a reject? Which I...might agree with...maybe?

Reply:
Hmm, I guess if we ignore, those messages won't get very far anyways, so it either doesn't really matter, or ignore is leaning on a slightly safer side when it comes to pubsub scoring.